### PR TITLE
486737: Test for Maven extensions (AbstractMavenLifecycleParticipant)

### PR DESCRIPTION
**Pull request by erwint**
A very simple extension to be tested is included and explicitly
whitelisted through the new extension point.

Bug: #486737
Change-Id: Id7f28a549f44afc419d430a151a85bb705805b7f

### CODE REVIEW

File: org.eclipse.m2e.tests/repositories/testrepo-src/486737_lifecycle-participant/src/main/java/org/eclipse/m2e/test/lifecycleparticipant/TestLifecycleParticipant.java [hunk 1]
```diff
@@ -0,0 +1,36 @@
+package org.eclipse.m2e.test.lifecycleparticipant;
```
Comment (fbricon) [inline org.eclipse.m2e.tests/repositories/testrepo-src/486737_lifecycle-participant/src/main/java/org/eclipse/m2e/test/lifecycleparticipant/TestLifecycleParticipant.java]: please add a copyright header

File: org.eclipse.m2e.tests/src/org/eclipse/m2e/tests/embedder/MavenImplTest.java [hunk 2]
```diff
@@ -559,4 +560,22 @@ public MavenExecutionResult call(IMavenExecutionContext context, IProgressMonito
 
     assertFalse(result.getExceptions().toString(), result.hasExceptions());
   }
+
+  public void test486737_readProject() throws Exception {
```
Comment (fbricon) [inline org.eclipse.m2e.tests/src/org/eclipse/m2e/tests/embedder/MavenImplTest.java]: this test seems superfluous compared to test486737_lifecycleParticipant

Comment (erwint) [inline org.eclipse.m2e.tests/src/org/eclipse/m2e/tests/embedder/MavenImplTest.java]: Igor requested tests both for MavenImpl #readProject and #readMavenProject, so here they are....

Comment (fbricon) [inline org.eclipse.m2e.tests/src/org/eclipse/m2e/tests/embedder/MavenImplTest.java]: scratch that, I see this it one of @ifedorenko 's requests

File: org.eclipse.m2e.tests/src/org/eclipse/m2e/tests/ProjectConfigurationManagerTest.java [hunk 3]
```diff
@@ -570,6 +570,37 @@ public void testHiddenFolderForArchetypeModule() throws Exception {
     }
   }
 
+  public void test486737_projectUpdate() throws Exception {
+    IProject project = importProject("projects/486737_lifecycleParticipant/pom.xml");
+    waitForJobsToComplete();
+    assertNoErrors(project);
```
Comment (fbricon) [inline org.eclipse.m2e.tests/src/org/eclipse/m2e/tests/ProjectConfigurationManagerTest.java]: I think it'd be better to read a pom not declaring the extension 1st, check source folders are src/main/java|resources, then switch the pom to activate the extension, update and check the new src folder is declared

**Comment by fbricon** [comment]
you also need to test when a project declares the extension first, source folders should be dummySrc right after project import. then change the pom to remove the extension and detect the source changed after project  update

**Comment by cpfeiffer** [comment]
Hi Fred,
thanks for your review -- I updated the tests accordingly.

**Comment by akurtakov** [comment]
Are you still interested in this one? Overall we prefer if tests are being added to https://github.com/eclipse-m2e/m2e-core for proper IP log and easier releng. Please consider opening one against this repository.

**Pull request closed without merging**